### ISSUE TITLE
Exact training and validation procedure

### ISSUE THREAD

**Issue by CanPhi** [jaredleekatzman/DeepSurv#44]
Hi, 

thank you very much for this github! I was wondering if you could provide me with a little more detail on how exactly you achieved the c-index reported in the paper. You mention doing a hyperparameter search with 3-fold cross-validation. I would like to benchmark your publication against one that I am soon going to publish. 

What did you do once the optimal hyperparameters were obtained? Did you still use a validation set for early stopping, or train until the mean early stopping epoch on the full training set? Or is the prediction an ensemble of the three models from the hyperparameter search?

If I understand your tutorials and code correctly, I cannot just use your docker files since I am modifying the code (adding a couple of line that require another hyperparameter). 

Also in your paper, you describe the WHAS data set as consisting of 5 features, but the HDF5 data set when downloaded seems to have 6 features. Am I getting something wrong or is that a typo in the paper? 

Also, did you simply use the default values in DeepSurv.train() for n_epochs, patience etc.? If not, where can I get those values from?

Thanks!
Philippe

**Comment by jaredleekatzman** [comment]
Hi Philippe,

Once I have the optimal training parameters, I train the network and do
early stopping if the validation accuracy does not improve by a certain
percentage.

I use a separate validation set for that early stopping and for validating
post trading that the tuned hyper-parameters worked.

If you modify DeepSurv to add hyper parameters, the docker files should
still be work (at least need a little modification). What's do your changes
look like such that isn't the case?

Jared

**Comment by jaredleekatzman** [comment]
Hi Philippe,

I actually copy over the entire contents of the Deep surv directory and
access the module locally. So as long as your changes are made in the
directory referenced by the docker file, it should work!

As for those additional parameters, they should be just using the default
values defined in the package. I can't remember if they are configurable
from the hyper parameters configuration.

Good luck!

Jared

**Comment by CanPhi** [comment]
Thanks, the code is up and running and producing the expected results! Turns out I vastly overestimated the importance of these parameters for the training, at least on WHAS it is really robust. Thanks again!
